environs/space/spaces.go

```diff
@@ -19,25 +19,15 @@ import (
 
 var logger = loggo.GetLogger("juju.environs.space")
 
-// Space represents a space that saved to state.
-type Space interface {
```

Comment:
main change here

Reply:
So in theory I'm fine with this change, but I have a sinking feeling something isn't quite right.

`SpaceInfo` isn't the same as `Space`. `Space` is a superset of `SpaceInfo`, so you can't represent `SpaceInfo` in the same places that `Space` can be represented. This is evident by the fact that you're now having to check the life of the space even though you already have that information. That information was thrown away when we went from a `State` to `StateInfo`. This problem will still exist once you move to dqlite and we keep life. This is all moot if we remove life.

Reply:
I entirely understand your feeling, but this change is just the scaffolding for what's coming (`life` is removed in the schema https://github.com/juju/juju/pull/16512). I think that once we apply this change and the new domain service is finished, the temporary methods will be cleaned up and everything should only use `SpaceInfo` (which should become `Space`) and it will be our only model in core.